pkg/environment/action_digest_subdirectory_manager.go

```diff
@@ -41,9 +39,14 @@ func (em *actionDigestSubdirectoryManager) Acquire(actionDigest *util.Digest) (M
 		return nil, err
 	}
 
-	// Create build directory within.
+	// Create build directory within. Only use a small number of
+	// characters from the digest to ensure the absolute path of the
+	// build directory remains short. This avoids reaching PATH_MAX
+	// and sockaddr_un::sun_path size limits for stronger digest
+	// functions. 16 characters is more than sufficient to prevent
+	// collisions.
 	buildDirectory := environment.GetBuildDirectory()
-	subdirectoryName := actionDigest.GetKey(em.subdirectoryFormat)
+	subdirectoryName := actionDigest.GetHashString()[:16]
```

Comment:
Yeah, 16 characters is more than sufficient. Even in the case of hash collisions there's no real problem. It will trigger a retriable error, meaning Bazel will simply retry it. On clusters with a higher number of workers, there is a slim chance of subsequent failures.